Source/BusinessLogic/Logic/PlayedGames/PlayedGameRetriever.cs

```diff
@@ -134,9 +138,9 @@ select gamingGroups
             return publicGameSummaries;
         }
 
-        public List<PlayedGameSearchResult> SearchPlayedGames(PlayedGameFilter playedGameFilter)
+        public List<PlayedGameSearchResult> SearchPlayedGames(PlayedGameFilter playedGameFilter, ApplicationUser currentUser)
```

Comment:
Now that I'm looking at this I think we should either do a null check on currentUser if they set the IncludeCurrentPlayerGamesOnly flag to true, and throw a new ArgumentNullException(nameof(currentUser)) and indicate that currentUser has to be set if this flag is true. I can do this if you are unsure or uninterested in making the change :)